src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,94 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.AlreadyRegisteredException;
+import core.basesyntax.exceptions.ValidDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws AlreadyRegisteredException, ValidDataException {
+        checkNulls(user);
+        isLoginNotRegistered(user);
+        isIdNotRegistered(user);
+        isPasswordLoginLengthValid(user);
+        isPasswordValid(user);
+        isLoginValid(user);
+        isAgeValid(user);
+        return storageDao.add(user);
+    }
+
+    private boolean checkNulls(User user) throws ValidDataException {
+        // Null login check
+        if (user.getLogin() == null) {
+            throw new ValidDataException("You did`t fill login field!");
+        }
+        // Null password check
+        if (user.getPassword() == null) {
+            throw new ValidDataException("You did`t fill password field!");
+        }
+        // Null age check
+        if (user.getAge() == null) {
+            throw new ValidDataException("You did`t fill age field!");
+        }
+        return true;
+    }
+
+    private boolean isLoginNotRegistered(User user) throws AlreadyRegisteredException {
+        // Same login check
+        if (storageDao.get(user.getLogin()) != null
+                && storageDao.get(user.getLogin()).getLogin().equals(user.getLogin())) {
```

Comment:
` && storageDao.get(user.getLogin()).getLogin().equals(user.getLogin())`
it's redundant